spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala

```diff
@@ -378,3 +379,12 @@ case class ColumnMappingPreDowngradeCommand(table: DeltaTableV2)
     true
   }
 }
+
+case class CheckConstraintsPreDowngradeTableFeatureCommand(table: DeltaTableV2)
+    extends PreDowngradeTableFeatureCommand {
+  override def removeFeatureTracesIfNeeded(): Boolean = {
```

Comment:
It would be nice to have a short comment here about what you explained above in https://github.com/delta-io/delta/pull/2987/files#r1635146168.

Reply:
Done!